### PR TITLE
Convert ProgramIDs from ids.ID to []byte to support size suffix for state keys

### PR DESCRIPTION
**Pull request by dboehm-avalabs**
Convert ProgramIDs from ids.ID to []byte to support size suffix for state keys

### CODE REVIEW

**Comment by richardpringle** [comment]
@dboehm-avalabs 
>to support size suffix for state keys

Is the size-suffix not a fixed size itself?

File: x/programs/rust/wasmlanche-sdk/src/program.rs [hunk 1]
```diff
@@ -120,7 +120,7 @@ impl Program {
     /// # Panics
     /// Panics if there was an issue deserializing the account
     #[must_use]
-    pub fn deploy(&self, program_id: Id, account_creation_data: &[u8]) -> Address {
+    pub fn deploy(&self, program_id: &[u8], account_creation_data: &[u8]) -> Address {
```
Comment (richardpringle) [inline x/programs/rust/wasmlanche-sdk/src/program.rs:123]: Can you just change the `Id` type instead? Then you can make this `&Id` (or the type could still be owned).

**Review by richardpringle — CHANGES_REQUESTED** [review]

File: x/programs/rust/wasmlanche-sdk/src/program.rs (hunk 1 above)
Comment (dboehm-avalabs) [reply x/programs/rust/wasmlanche-sdk/src/program.rs:123]: it would have to be a distinct type since IDs still exist as the result of hashes.

**Comment by dboehm-avalabs** [comment]
> @dboehm-avalabs
> 
> > to support size suffix for state keys
> 
> Is the size-suffix not a fixed size itself?

In this case, yes they are, but thought it might be worthwhile to support program ids that aren't to allow more flexibility to VM creators.  Having them be fixed size does have advantages though performance wise, so something I want to think about more in the future.

**Pull request closed without merging**